Modules/Loadable/Markups/MRML/vtkCurveMeasurementsCalculator.cxx

```diff
@@ -314,6 +326,176 @@ bool vtkCurveMeasurementsCalculator::CalculatePolyDataCurvature(vtkPolyData* pol
+    return false;
+    }
+
+  vtkCellArray* lines = polyData->GetLines();
+  vtkNew<vtkIdList> linePoints;
+
+  lines->GetCell(0, linePoints);
+  vtkIdType numberOfPoints = // Last point in closed curve line is the first point
+    (this->CurveIsClosed ? linePoints->GetNumberOfIds()-1 : linePoints->GetNumberOfIds());
+
+  // Initialize torsion array
+  vtkNew<vtkDoubleArray> torsionArray;
+  torsionArray->Initialize();
+  torsionArray->SetName("Torsion");
+  torsionArray->SetNumberOfComponents(1);
+  torsionArray->SetNumberOfTuples(numberOfPoints);
+  torsionArray->Reset();
+  torsionArray->FillComponent(0,0.0);
+
+  // Initialize curvature variables
+  double minTorsion= 0.0;
+  double maxTorsion = 0.0;
+  double meanTorsion = 0.0;
+
+  // Get values for first point
+  double* binormal = binormals->GetTuple3(linePoints->GetId(1));
+  double binormalNorm = sqrt(binormal[0]*binormal[0] + binormal[1]*binormal[1] + binormal[2]*binormal[2]);
+  double normBinormal[3] = {0.0, 0.0, 0.0};
+  double prevNormBinormal[3] = {binormal[0]/binormalNorm, binormal[1]/binormalNorm, binormal[2]/binormalNorm};
+  double torsion = 0.0;
+  double *tangent = nullptr;
+  double currentLength = 0.0;
+  double length = 0.0;
+
+  // The torsion for the first cell is 0.0 for open curves
+  torsionArray->InsertValue(linePoints->GetId(0), 0.0);
+
+  for (vtkIdType idx=1; idx<numberOfPoints-1; ++idx)
+    {
+    binormal = binormals->GetTuple3(linePoints->GetId(idx+1));
+    binormalNorm = sqrt(binormal[0]*binormal[0] + binormal[1]*binormal[1] + binormal[2]*binormal[2]);
+
+    normBinormal[0] = binormal[0] / binormalNorm;
+    normBinormal[1] = binormal[1] / binormalNorm;
+    normBinormal[2] = binormal[2] / binormalNorm;
+
+    // Local torsion
+    torsion = sqrt( (normBinormal[0]-prevNormBinormal[0])*(normBinormal[0]-prevNormBinormal[0])
+                  + (normBinormal[1]-prevNormBinormal[1])*(normBinormal[1]-prevNormBinormal[1])
+                  + (normBinormal[2]-prevNormBinormal[2])*(normBinormal[2]-prevNormBinormal[2]) )
+              / binormalNorm;
+    torsionArray->InsertValue(linePoints->GetId(idx), torsion);
+
+    // Statistics
+    tangent = tangents->GetTuple3(linePoints->GetId(idx));
+    currentLength = sqrt( tangent[0]*tangent[0] + tangent[1]*tangent[1] + tangent[2]*tangent[2] );
+    if (torsion < minTorsion)
+      {
+      minTorsion = torsion;
+      }
+    else if (torsion > maxTorsion)
+      {
+      maxTorsion = torsion;
+      }
+    meanTorsion += torsion * currentLength; // weighted mean
+    length += currentLength;
+
+    // Propagate current values to previous
+    for (int i=0; i<3; ++i)
+      {
+      prevNormBinormal[i] = normBinormal[i];
+      }
+    } // For each line point
+
+  if (!this->CurveIsClosed)
+    {
+    // The torsion for the last cell by definition is 0.0 for open curves
+    torsionArray->InsertValue(linePoints->GetId(numberOfPoints-1), 0.0);
```

Comment:
We should copy the previous distortion value, because that is a better estimation than 0.